client/js/socket-events/msg.js

```diff
@@ -70,12 +70,25 @@ function processReceivedMessage(data) {
 
 	notifyMessage(targetId, channel, data);
 
-	const lastVisible = container.find("div:visible").last();
+	let shouldMoveMarker = data.msg.self;
 
-	if (data.msg.self
-		|| lastVisible.hasClass("unread-marker")
-		|| (lastVisible.hasClass("date-marker")
-		&& lastVisible.prev().hasClass("unread-marker"))) {
+	if (!shouldMoveMarker) {
+		const lastChild = container.children().last();
+
+		// If last element is hidden (e.g. hidden status messages) check the element before it.
+		// If it's unread marker or date marker, then move unread marker to the bottom
+		// so that it isn't displayed as the last element in chat.
+		// display properly is checked instead of using `:hidden` selector because it doesn't work in non-active channels.
+		if (lastChild.css("display") === "none") {
```

Comment:
I'm testing this and it looks fine, but will this be buggy when changing the settings for condensed messages, until the next app reload?

Reply:
That always was the case. New message or switching channel would fix it.

Reply:
I didn't realize that was the case on `master`. Workarounds that involve reloading the app or switching channels are unfortunate, but since it's not a breakage you introduced here, that's fine to ignore for now.